matisse/src/main/java/com/zhihu/matisse/internal/loader/AlbumMediaLoader.java

```diff
@@ -73,6 +89,28 @@ private AlbumMediaLoader(Context context, Uri uri, String[] projection, String s
 
     public static CursorLoader newInstance(Context context, Album album, boolean capture) {
         if (album.isAll()) {
+            SelectionSpec selectionSpec = SelectionSpec.getInstance();
+
+            if (restrictedToImages(selectionSpec)) {
+                return new AlbumMediaLoader(
+                        context,
+                        QUERY_URI,
+                        PROJECTION,
+                        SELECTION_ALL,
+                        SELECTION_IMAGE_ARGS,
+                        ORDER_BY,
+                        capture);
+            } else if (restrictedToVideos(selectionSpec)) {
+                return new AlbumMediaLoader(
+                        context,
+                        QUERY_URI,
+                        PROJECTION,
+                        SELECTION_ALL,
+                        SELECTION_VIDEO_ARGS,
+                        ORDER_BY,
+                        capture);
+            }
+
```

Comment:
I think the modifications you made are not rigorous. `SELECTION_ALL` are not matched with `SELECTION_*_ARGS` when `restrictedTo*s()` returns true.   
And also, all the other album pages except ALL, still show up both images and videos.